### PR TITLE
ansible: improve Python 3 compatibility

### PR DESCRIPTION
**Pull request by cclauss**
As requested at https://github.com/nodejs/build/pull/1399#issuecomment-522687404, this PR cherrypicks key modifications from #1399 to improve the Python 3 compatibility of our build process.
* __git checkout refack/pipenv+python3 -- .flake8 .gitignore ansible/plugins/library/remmina_config.py ansible/plugins/library/ssh_config.py setup/www/tools/metrics/country-lookup.py jenkins/scripts/coverage/generate-index-html.py__
* __code ansible/plugins/inventory/nodejs_yaml.py__
    * cherrypick some change but not others
    * sync __valid["arch"]__ with https://github.com/nodejs/node/blob/master/configure.py
* __flake8 .__  # placate flake8

### CODE REVIEW

File: ansible/plugins/inventory/nodejs_yaml.py [hunk 1]
```diff
@@ -23,35 +23,32 @@
...
-  # valid roles - add as necessary
-  'type': ('infra', 'release', 'test'),
-
-  # providers - validated for consistency
-  'provider': ('azure', 'digitalocean', 'joyent', 'ibm', 'linuxonecc',
-               'macstadium', 'marist', 'mininodes', 'msft', 'osuosl',
-               'rackspace', 'requireio', 'scaleway', 'softlayer', 'voxer',
-               'packetnet', 'nearform')
+    # taken from nodejs/node.git: ./configure
+    'arch': ('arm', 'arm64', 'ia32', 'mips', 'mipsel', 'mips64el', 'ppc',
```
Comment (sam-github) [inline ansible/plugins/inventory/nodejs_yaml.py]: you added arm and x86_64? that doesn't seem related to py3 compat

File: jenkins/scripts/coverage/generate-index-html.py [hunk 2]
```diff
@@ -6,9 +6,9 @@
 with open('out/index.csv') as index:
   index_csv = filter(lambda line: line, index.read().split('\n'))
 
+# noqa
```
Comment (sam-github) [inline jenkins/scripts/coverage/generate-index-html.py]: what does this mean?

File: .flake8 [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[flake8]
+exclude=.venv
+max-line-length=250
+import-order-style=google
+ignore=E111
+
```
Comment (sam-github) [inline .flake8]: trailing empty lines can be removed

File: ansible/plugins/inventory/nodejs_yaml.py (hunk 1 above)
Comment (cclauss) [reply ansible/plugins/inventory/nodejs_yaml.py]: Reverted

File: jenkins/scripts/coverage/generate-index-html.py (hunk 2 above)
Comment (cclauss) [reply jenkins/scripts/coverage/generate-index-html.py]: Reverted.  https://github.com/nodejs/build/pull/1399#discussion_r266726111 but we no longer need it.

File: .flake8 (hunk 3 above)
Comment (cclauss) [reply .flake8]: Fixed.

**Comment by sam-github** [comment]
Out of curiosity, is flake8 run by anything in our  tooling? I'd never heard of it till this PR.

**Comment by cclauss** [comment]
* #1930
* https://github.com/nodejs/node/blob/master/Makefile#L1353
* https://github.com/nodejs/node-gyp/blob/master/.travis.yml#L68
